src/services/api/routes/communities.ts

```diff
@@ -284,4 +302,122 @@ router.post('/', multer().none(), async function (request: express.Request, resp
 	}).end({ pretty: true, allowEmpty: true }));
 });
 
+router.post('/:community_action', multer().none(), async function (request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
+	if (!parentCommunity) {
+		await respondCommunityNotFound(response);
+		return;
+	}
+
+	const communityActions: string[] = request.params.community_action.split('.');
+	const community: HydratedCommunityDocument | null = await commonGetSubCommunity(request, response, communityActions[0]);
+	if (!community) {
+		return;
+	}
+
+	if (communityActions.length > 1) {
+		if (communityActions[1] === 'delete') {
+			await deleteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'favorite') {
+			await addFavoriteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'unfavorite') {
+			await delFavoriteSubCommunity(community, request, response);
+		} else { // '%s.search'
+			response.sendStatus(501); // Not Implemented
+		}
+	} else {
+		await updateSubCommunity(community, request, response);
+	}
+});
```

Comment:
This is for the `/v1/communities/%s.action` handlers correct? Express supports partial route parameters, so I wonder if it would be more clear to actually define these as `/:community_id.delete` and so on. That way you don't have to look at a single route handlers body to know which actions are supported, you can instead see the individual routes that are defined to know which actions we have. It also makes it more clear what the URLs structure is since right now it's not immediately clear that `community_action` actually has the format `community_id.action`. The functions you made for each action are only ever used once, so it would remove the need to call an extra function as well